torch_xla/csrc/tensor_methods.cpp

```diff
@@ -1876,7 +1876,7 @@ std::pair<XLATensorPtr, XLATensorPtr> XLATensor::nms(
 XLATensorPtr XLATensor::nonzero(const XLATensorPtr& input) {
   torch::lazy::NodePtr node =
       torch::lazy::MakeNode<NonZero>(input->GetIrValue());
-  return input->CreateFrom(torch::lazy::Value(node, 0), at::ScalarType::Long);
```

Comment:
I wonder why we removed the `at::ScalarType::Long`?

Reply:
this way we don't overwrite the dtype of the result of the nonzero